common/network-common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java

```diff
@@ -206,6 +207,11 @@ public void onFailure(Throwable e) {
   private void processOneWayMessage(OneWayMessage req) {
     try {
       rpcHandler.receive(reverseClient, req.body().nioByteBuffer());
+    } catch (InvalidClassException ice) {
```

Comment:
The problem is that this won't catch all or even most errors resulting from version incompatibility. Spark has never supported or contemplated mis-matching versions internally. I don't think we should try to handle this, because it's fundamentally piecemeal and hacky.

Reply:
Perhaps I should have picked different name for the PR than `Handling spark-submit and master version mismatch`. It doesn't try to solve the issue in a complex way that two different version could be able to talk to each other, all it does is saying the user. Hey, you have probably different version than spark master. I agree, it's little bit hacky, on the other hand I see no other option than to catch the `InvalidClassException`, if the version is not part of the message. Perhaps some initial handshake in which the version is sent would be cleaner.

What about re-throwing the exception. This way it wouldn't change the semantics of the code, but the client would be informed. wdyt?